src/ring_buffer.c

```diff
@@ -0,0 +1,139 @@
+/**
+ * Race Capture Pro Firmware
+ *
+ * Copyright (C) 2015 Autosport Labs
+ *
+ * This file is part of the Race Capture Pro fimrware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mem_mang.h"
+#include "mod_string.h"
+#include "ring_buffer.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+
+static size_t get_end_dist(struct ring_buff *rb, const char *p)
```

Comment:
If I did that we should do it separately.  And also probably come up with a common trait for this so it can apply consistently throughout.  But I think its a good idea.